bindings/src/main/java/io/pravega/storage/gcp/GCPChunkStorage.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Override
+    protected void doSetReadOnly(ChunkHandle handle, boolean isReadOnly) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException("GCP chunk storage does not support doSetReadOnly");
+    }
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws ChunkNotFoundException {
+        Blob blob = this.storage.get(this.config.getBucket(), getObjectPath(chunkName), Storage.BlobGetOption.fields(Storage.BlobField.SIZE));
+        if (null == blob) {
+            throw new ChunkNotFoundException(chunkName, "doGetInfo", null);
+        }
+        return ChunkInfo.builder()
+                .name(chunkName)
+                .length(blob.getSize())
+                .build();
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) {
+        throw new UnsupportedOperationException("GCPChunkStorage does not support creating object without content.");
+    }
+
+    @Override
+    protected ChunkHandle doCreateWithContent(String chunkName, int length, InputStream data) throws ChunkStorageException {
+        BlobId blobId = BlobId.of(this.config.getBucket(), getObjectPath(chunkName));
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("application/octet-stream").build();
+        try (InputStream inputStream = data) {
+            this.storage.createFrom(blobInfo, inputStream);
+            return ChunkHandle.writeHandle(chunkName);
+        } catch (IOException e) {
+            throw convertException(chunkName, "doCreateWithContent", e);
+        } catch (Exception e) {
+            throw convertException(chunkName, e.getMessage(), new ChunkStorageException(e.getMessage(), "doCreateWithContent", null));
+        }
+    }
+
+    @Override
+    protected boolean checkExists(String chunkName) {
+        Page<Blob> blobs = this.storage.list(this.config.getBucket(), Storage.BlobListOption.prefix(getObjectPath(chunkName)), Storage.BlobListOption.pageSize(1));
+        return blobs.getValues().iterator().hasNext();
+    }
+
+    @Override
+    protected void doDelete(ChunkHandle handle) throws ChunkNotFoundException {
+        boolean deleted = this.storage.delete(this.config.getBucket(), getObjectPath(handle.getChunkName()));
+        if (!deleted) {
+            throw new ChunkNotFoundException(handle.getChunkName(), "doDelete", null);
+        }
+    }
+
+    /**
+     *
+     * @param chunkName - Name of the chunk
+     * @param message - Error message
+     * @param e - Exception
+     * @return Takes the exception and converts it to meaningful exception
+     */
+    private ChunkStorageException convertException(String chunkName, String message, Exception e) {
+        ChunkStorageException retValue = null;
+        if (e instanceof ChunkStorageException) {
+            retValue = (ChunkStorageException) e;
+        } else if (e instanceof ChunkNotFoundException) {
+            retValue = (ChunkNotFoundException) e;
```

Comment:
This else block is never going to be executed because ChunkNotFoundException 
 inherits from ChunkStorageException.

Just delete 
 ```
else if (e instanceof ChunkNotFoundException) {
            retValue = (ChunkNotFoundException) e;
        }
```

Reply:
This will also increase code coverage just by a notch. but probably enough to pass the code coverage requirement.
If not then add some tests for doWrite